src/libsync/networkjobs.cpp

```diff
@@ -344,32 +384,60 @@ void LsColJob::start()
     AbstractNetworkJob::start();
 }
 
-// TODO: Instead of doing all in this slot, we should iteratively parse in readyRead(). This
-// would allow us to be more asynchronous in processing while data is coming from the network,
-// not all in one big blob at the end.
-bool LsColJob::finished()
+void LsColJob::newReplyHook(QNetworkReply *reply)
 {
-    qCInfo(lcLsColJob) << "LSCOL of" << reply()->request().url() << "FINISHED WITH STATUS"
-                       << replyStatusString();
+    qDebug() << Q_FUNC_INFO << reply;
+    connect(reply, &QNetworkReply::metaDataChanged, this, &LsColJob::slotMetaDataChanged);
+    connect(reply, &QIODevice::readyRead, this, &LsColJob::slotReadyRead);
+    // FIXME networkActivity for timeout reset?
+}
 
+void LsColJob::slotMetaDataChanged() {
     QString contentType = reply()->header(QNetworkRequest::ContentTypeHeader).toString();
     int httpCode = reply()->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
     if (httpCode == 207 && contentType.contains("application/xml; charset=utf-8")) {
-        LsColXMLParser parser;
-        connect(&parser, &LsColXMLParser::directoryListingSubfolders,
+        // Initialize parser
+        QString expectedPath = reply()->request().url().path(); // something like "/owncloud/remote.php/webdav/folder"
+        _parser.initialize(expectedPath);
+
+        connect(&_parser, &LsColXMLParser::directoryListingSubfolders,
             this, &LsColJob::directoryListingSubfolders);
-        connect(&parser, &LsColXMLParser::directoryListingIterated,
+        connect(&_parser, &LsColXMLParser::directoryListingIterated,
             this, &LsColJob::directoryListingIterated);
-        connect(&parser, &LsColXMLParser::finishedWithError,
+        connect(&_parser, &LsColXMLParser::finishedWithError,
             this, &LsColJob::finishedWithError);
-        connect(&parser, &LsColXMLParser::finishedWithoutError,
+        connect(&_parser, &LsColXMLParser::finishedWithoutError,
             this, &LsColJob::finishedWithoutError);
+    }
+}
 
-        QString expectedPath = reply()->request().url().path(); // something like "/owncloud/remote.php/webdav/folder"
-        if (!parser.parse(reply()->readAll(), &_sizes, expectedPath)) {
-            // XML parse error
+void LsColJob::slotReadyRead() {
+    qDebug() << Q_FUNC_INFO;
+    if (_parser.isInitialized()) {
+        QByteArray snippet = reply()->readAll();
```

Comment:
Note: there is a function QXmlStreamReader::setDevice you probably should use that instead of using slotReadyRead and putting the whole buffer in a QByteArray.